src/result/ResultVariable.php

```diff
@@ -38,16 +47,32 @@ class ResultVariable implements \JsonSerializable
      * @param array $array
      * @return \oat\libCat\result\ResultVariable
      */
-    public static function restore($array)
+    public static function restore($array, $variableType = self::OUTCOME_VARIABLE)
```

Comment:
Serialization data alone should be sufficient to restore the objects, no outside information should be required